### PR TITLE
chore(span)!: remove Span.meta

### PR DESCRIPTION
**Pull request by mabdinur**
BLOCKED BY: #3212

#3212 will deprecate Span.meta on master

After #3212 is merged and 1.0-dev is rebased this PR will remove the Span.meta property

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

File: ddtrace/ext/__init__.py
Comment (brettlangdon) [inline ddtrace/ext/__init__.py:1]: unintended change?

Comment (mabdinur) [reply ddtrace/ext/__init__.py]: Yup, I made the target branch master instead of 1.0-dev

File: ddtrace/internal/runtime/runtime_metrics.py
Comment (Kyle-Verhoog) [inline ddtrace/internal/runtime/runtime_metrics.py:99]:
```suggestion
            span._meta["language"] = "python"
```

File: ddtrace/span.py
Comment (Kyle-Verhoog) [inline ddtrace/span.py:343]: Not a fan of this new api since it returns a mutable reference to the underlying tags data structure.

eg.

```python
tags = span.get_tags()
tags["new"] = "foo"
assert span.get_tag("new") == "foo"
```

maybe a better approach is an iterator or generator over all the tags stored in meta?

File: ddtrace/contrib/wsgi/wsgi.py
Comment (Kyle-Verhoog) [inline ddtrace/contrib/wsgi/wsgi.py:130]:
```suggestion
                        resp_span._meta["result_class"] = resp_class
```

have to avoid using `set_tag` unfortunately

**Review by Kyle-Verhoog — CHANGES_REQUESTED** [review]

File: ddtrace/contrib/aiobotocore/patch.py
Comment (majorgreys) [inline ddtrace/contrib/aiobotocore/patch.py:59]: Can we change to not use the internalized meta (with an addition of a `get_tags`)?

```
span.set_tags(self._self_parent_span.get_tags())
```

File: ddtrace/contrib/wsgi/wsgi.py
Comment (majorgreys) [inline ddtrace/contrib/wsgi/wsgi.py:130]: Also indirection here:

```suggestion
                        resp_span.set_tag("result_class", resp_class)
```

**Review by majorgreys — COMMENTED** [review]
I would also avoid accessing `_meta` in tests, by using `get_tag` and `get_tags` (to be added) where needed. The other option would be to change it to `to_dict()["meta"]`.

File: releasenotes/notes/internalize-span-meta-d557d84d3764c4c8.yaml
Comment (Kyle-Verhoog) [inline releasenotes/notes/internalize-span-meta-d557d84d3764c4c8.yaml:4]:
```suggestion
    Internalize Span.meta. Use ``Span.get_tag(key)`` and ``Span.set_tag(key, value)`` to update tags.
```

**Comment by mabdinur** [comment]
BLOCKED BY: https://github.com/DataDog/dd-trace-py/pull/3212 

#3212 will deprecate Span.meta on master 

After #3212 is merged and 1.0-dev is rebased this PR will remove the Span.meta property

File: ddtrace/contrib/aiobotocore/patch.py
Comment (Kyle-Verhoog) [reply ddtrace/contrib/aiobotocore/patch.py:59]: I'm ok with this for now but it should change in the future, there is some other weird stuff happening here with `resource`, `span_type` and `metrics`... Not sure why we're copying them at all.

File: ddtrace/contrib/wsgi/wsgi.py
Comment (Kyle-Verhoog) [reply ddtrace/contrib/wsgi/wsgi.py:130]: Could maybe do `_set_str_tag` instead which is a bit more established as an internal API.

**Review by Kyle-Verhoog — COMMENTED** [review]
Looks good to me once the master PR is merged!

**Pull request closed without merging**
